prompting/tasks/date_qa.py

```diff
@@ -2,34 +2,46 @@
 from prompting.tasks import Task
 from prompting.cleaners.cleaner import CleanerPipeline
 
-
-SECTION_MESSAGES = {"Births": " was born ", "Deaths": " died ", "Events": " "}
-
+QUERY_SYSTEM_PROMPT = """You are a question creation expert. When asked to create a question, you use the context to make a specific question that would have the answer <date>. Your question should contain the topic."""
+QUERY_PROMPT_TEMPLATE = """\
+Create a question about {topic} that would have <date> as the answer using the following context:
+topic: {topic}
+context: {context}
+"""
+REFERENCE_PROMPT_TEMPLATE = """\
+Your answer must include the following date: {date}.
+Answer the following question using the provided context. 
+Question: {query}
+Context: {context}
+"""
 
 @dataclass
 class DateQuestionAnsweringTask(Task):
     name = "date_qa"
+    challenge_type = 'query'
     desc = "get help answering a specific date-based question"
     goal = "to get the answer to the following date-based question"
     reward_definition = [
-        dict(name="date", weight=1.0),
+        dict(name="date", weight=0.7),
+        dict(name="rouge", weight=0.3),
     ]
     penalty_definition = []
     cleaning_pipeline = [
-        dict(name="remove_quotes"),
+        #dict(name="remove_quotes"),
         dict(name="remove_roles"),
+        dict(name="remove_tags"), 
+        dict(name="first_question"),
     ]
-    static_reference = True
-    static_query = True
+    static_reference = False
 
     def __init__(self, llm_pipeline, context, create_reference =True):
         self.context = context
-
-        self.query = (
-            context.content + SECTION_MESSAGES[context.topic] + "on what exact date?"
-        )
-        self.reference = self.context.title.replace("_", " ") + ", " + context.subtopic
-
+        self.query_system_prompt = QUERY_SYSTEM_PROMPT
+        self.query_prompt = QUERY_PROMPT_TEMPLATE.format(topic = context.title, context=context.content)
+        self.query = self.generate_query(llm_pipeline)
+        date = self.context.extra.get('date', None)
+        self.reference_prompt = REFERENCE_PROMPT_TEMPLATE.format(date = date, query = self.query, context = context.content)
+        self.reference = self.generate_reference(llm_pipeline, clean = False).replace('?',"")
```

Comment:
I think we need to add the condition of 
```python
 if create_reference:
        self.reference = self.generate_reference(llm_pipeline)
```
so we can parallelize the reference generation at a later step, similar to the implementation of the others llm based tasks